subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskDependency.java

```diff
@@ -40,23 +41,32 @@
 import static org.gradle.util.GUtil.uncheckedCall;
 
 public class DefaultTaskDependency extends AbstractTaskDependency {
-    private Set<Object> values;
+    private final ImmutableSet<Object> immutableValues;
+    private Set<Object> mutableValues;
     private final TaskResolver resolver;
 
     public DefaultTaskDependency() {
         this(null);
     }
 
     public DefaultTaskDependency(TaskResolver resolver) {
+        this(resolver, ImmutableSet.of());
+    }
+
+    public DefaultTaskDependency(TaskResolver resolver, ImmutableSet<Object> tasks) {
```

Comment:
`ImmutableSet<Object> tasks` -> the name _tasks_ can be confusing here. I think _values_ (as used elsewhere in this class) would be better.